### PR TITLE
docs(ReactNative): improve example

### PR DESCRIPTION
**Pull request by mthuret**
This PR improve our React Native example by featuring a complete application. 

Next step here is to create a Getting Started guide about the concept you need to know in order to create this app. 

Here's a GIF if you don't want to run the app locally :)

react-native-example-final

### CODE REVIEW

**Comment by marielaures** [comment]
I think it is really great @mthuret!
Maybe I would add some way to let the user know that the filters have been modified because once the user is on the result pages there is no indication that the filters have been modified!

File: packages/react-instantsearch/examples/react-native/.gitignore [hunk 1]
```diff
@@ -34,9 +34,20 @@ local.properties
...
 *.keystore
+
+# fastlane
+#
+# It is recommended to not store the screenshots in the git repo. Instead, use fastlane to re-generate the
+# screenshots whenever they are needed.
+# For more information about the recommended setup visit:
+# https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Gitignore.md
+
+fastlane/report.xml
```
Comment (vvo) [inline packages/react-instantsearch/examples/react-native/.gitignore]: What do we use fastlane for (newbe here)?

File: packages/react-instantsearch/examples/react-native/android/app/src/main/java/com/reactnative/MainApplication.java [hunk 2]
```diff
@@ -17,14 +16,15 @@
 
   private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
     @Override
-    protected boolean getUseDeveloperSupport() {
```
Comment (vvo) [inline packages/react-instantsearch/examples/react-native/android/app/src/main/java/com/reactnative/MainApplication.java]: Are those changes automated by create native app or something?

File: packages/react-instantsearch/examples/react-native/ios/reactnative/Info.plist [hunk 3]
```diff
@@ -1,68 +1,72 @@
...
-           <string>Ionicons.ttf</string>
-           <string>MaterialIcons.ttf</string>
-           <string>Octicons.ttf</string>
-           <string>SimpleLineIcons.ttf</string>
-           <string>Zocial.ttf</string>
-         </array>
-</dict>
-</plist>
+  <dict>
+    <key>CFBundleDevelopmentRegion</key>
```
Comment (vvo) [inline packages/react-instantsearch/examples/react-native/ios/reactnative/Info.plist]: so, much, boilerplate :D

File: packages/react-instantsearch/examples/react-native/src/Home.js [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+  searchState: PropTypes.object,
+};
+
+export default Home;
+
+class SearchBox extends Component {
+  render() {
+    return (
+      <View style={styles.searchBoxContainer}>
+        <Spinner left={60} />
```
Comment (vvo) [inline packages/react-instantsearch/examples/react-native/src/Home.js]: yeah spinner

File: packages/react-instantsearch/package.json [hunk 5]
```diff
@@ -30,6 +30,7 @@
     "react": "^15.5.4",
     "react-dom": "^15.5.4",
     "react-native": "^0.41.2",
-    "react-test-renderer": "^15.5.4"
+    "react-test-renderer": "^15.5.4",
+    "react-navigation": "latest"
```
Comment (vvo) [inline packages/react-instantsearch/package.json]: isn't this dangerous?

Comment (vvo) [inline packages/react-instantsearch/package.json]: Also does it needs to be here or maybe in the example itself?

File: packages/react-instantsearch/examples/react-native/.gitignore (hunk 1 above)
Comment (mthuret) [inline packages/react-instantsearch/examples/react-native/.gitignore]: All of this files (and same below) are created by the react-native command line. I will probably switch to an expo directory before merging this PR. It's less files, and its easier to have the app running on your phone.

File: packages/react-instantsearch/package.json (hunk 5 above)
Comment (mthuret) [inline packages/react-instantsearch/package.json]: "react-navigation": "latest" => will disappear we are not using it anymore :) Thanks for catching it.

**Comment by mthuret** [comment]
We now use `create-react-native-app` along with `expo`. It's a lot more easier to run this app, especially on a real device. And it's also clearer as there are less files.

@marielaures I also added the filter count :)

@vvo, @spinach, @marielaures : give it a try and let me know how it goes.

File: packages/react-instantsearch/examples/react-native/src/Home.js [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+          apiKey="6be0576ff61c053d5f9a3225e2a90f76"
+          indexName="ikea"
+          searchState={this.state.searchState}
+          onSearchStateChange={this.onSearchStateChange}
+        >
+          <StatusBar backgroundColor="blue" barStyle="light-content" />
+          <ConnectedSearchBox />
+
+          <View style={styles.options}>
+            <ConnectedStats />
```
Comment (spinach) [inline packages/react-instantsearch/examples/react-native/src/Home.js]: @marielaures just a nitpicky thing: can you add left-padding for the stats label in the first screen?

**Comment by vvo** [comment]
confirming, expo is dope and it was flawless to run the app locally, gg

**Comment by MaganAnkur** [comment]
@mthuret 
could you please suggest what is the best way if we want to remove filters on clearing up the search.
i.e.  
- User search for apple 
- set few filters
- clear search query
- search for new query
- filters are still there

Any way we can clear up all the filters as soon as user clears up the query

**Comment by Haroenv** [comment]
@MaganAnkur you can make the search box into:

```jsx
const Clear = connectCurrentRefinements(({ refine, onClick }) => (
  <button
    onClick={() => {
      refine([]);
      onClick();
    }}
  >
    clear
  </button>
));

const SearchBox = connectSearchBox(({ refine, currentRefinement }) => (
  <>
    <input value={currentRefinement} onChange={(e) => refine(e.target.value)} />
    <Clear onClick={() => refine('')} />
  </>
));
```

(approximately, as I don't have a react native example available, the idea is the same as for dom)

**Comment by MaganAnkur** [comment]
Thanks @Haroenv I implemented in exact same way. Problem I am facing with this solution is, clear button is giving delayed response.
Pls look at my code below:


`const SearchScreenInput = ({ refine, currentRefinement }) => {
    const [searchQuery, setSearchQuery] = useState(currentRefinement);
    const [isInputFocussed, setInputFocussed] = useState(false);

    const onClear = () => {
        setSearchQuery('');
    };
    useEffect(() => {
        if (!searchQuery) refine('');
    }, [searchQuery]);

    const placeholder = isInputFocussed
        ? translate('searchScreen.placeholder.focused')
        : translate('searchScreen.placeholder.default');

    const headerBorder = isInputFocussed ? 'border-teal' : 'border-grey';

    const onFocus = () => {
        setInputFocussed(true);
    };

    const onBlur = () => {
        setInputFocussed(false);
    };

    return (
        <>
            <View style={tailwind(`flex-row flex-1 border-1  h-12 rounded-md mr-3 overflow-hidden px-3 ${headerBorder}`)}>
                <SearchIcon style={tailwind('self-center')} width={20} height={20} />
                <TextInput
                    style={[
                        tailwind(`text-sm text-grey-dark h-full mx-1 w-full flex-1`),
                        { fontFamily: 'Poppins-Regular', lineHeight: 17 },
                    ]}
                    onChangeText={setSearchQuery}
                    onSubmitEditing={({ nativeEvent: { text } }) => {
                        refine(text);
                    }}
                    onFocus={onFocus}
                    onBlur={onBlur}
                    value={searchQuery}
                    placeholder={placeholder}
                    returnKeyType="search"
                    autoCorrect={false}
                    placeholderTextColor={tailwind('text-grey-dark').color}
                />
                {searchQuery ? <ClearButton onClear={onClear}  /> : null}
            </View>
        </>
    );
};`

`const ClearButton = connectCurrentRefinements((props) => {
    const { onClear = () => {}, refine, canRefine, items } = props;
console.log(props)
    return (
        <TouchableOpacity
            style={tailwind('self-center')}
            onPress={() => {
                canRefine && refine(items);
                onClear();
               
            }}>
            <RoundCross />
        </TouchableOpacity>
    );
});`

so when on clear function executes two things are happening setting state and refine
which in turn take noticeable time to clear the query

**Pull request merged**